src/Components/Components/test/RendererTest.cs

```diff
@@ -4170,6 +4170,241 @@ public async Task ThrowsIfComponentProducesInvalidRenderTree()
+        {
+            // Arrange
+            var renderer = new TestRenderer();
+            Exception exception = null;
+            var rootComponent = new TestComponent(builder =>
+            {
+                TestErrorBoundary.RenderNestedErrorBoundaries(builder, builder =>
+                {
+                    builder.OpenComponent<ErrorThrowingComponent>(0);
+                    builder.AddAttribute(1, nameof(ErrorThrowingComponent.ThrowDuringParameterSettingSync), exception);
+                    builder.CloseComponent();
+                });
+            });
+            var rootComponentId = renderer.AssignRootComponentId(rootComponent);
+            renderer.RenderRootComponent(rootComponentId);
+            var errorBoundaries = renderer.Batches.Single().GetComponentFrames<TestErrorBoundary>()
+                .Select(f => (TestErrorBoundary)f.Component);
+            var errorThrowingComponentId = renderer.Batches.Single()
+                .GetComponentFrames<ErrorThrowingComponent>().Single().ComponentId;
+
+            // Act
+            exception = new InvalidTimeZoneException("Error during SetParametersAsync");
+            rootComponent.TriggerRender();
+
+            // Assert
+            Assert.Equal(2, renderer.Batches.Count);
+            Assert.Collection(errorBoundaries,
+                component => Assert.Null(component.ReceivedException),
+                component => Assert.Same(exception, component.ReceivedException));
+
+            // The failed subtree is disposed
+            Assert.Equal(errorThrowingComponentId, renderer.Batches[1].DisposedComponentIDs.Single());
+        }
+
+        [Fact]
+        public async Task SetParametersAsyncExceptionsCanBeHandledByClosestErrorBoundary_Async()
+        {
+            // Arrange
+            var renderer = new TestRenderer();
+            Exception exception = null;
+            var rootComponent = new TestComponent(builder =>
+            {
+                TestErrorBoundary.RenderNestedErrorBoundaries(builder, builder =>
+                {
+                    builder.OpenComponent<ErrorThrowingComponent>(0);
+                    builder.AddAttribute(1, nameof(ErrorThrowingComponent.ThrowDuringParameterSettingAsync), exception);
+                    builder.CloseComponent();
+                });
+            });
+            var rootComponentId = renderer.AssignRootComponentId(rootComponent);
+            renderer.RenderRootComponent(rootComponentId);
+            var errorBoundaries = renderer.Batches.Single().GetComponentFrames<TestErrorBoundary>()
+                .Select(f => (TestErrorBoundary)f.Component).ToArray();
+            var errorThrowingComponentId = renderer.Batches.Single()
+                .GetComponentFrames<ErrorThrowingComponent>().Single().ComponentId;
+
+            // Act/Assert 1: No synchronous errors
+            exception = new InvalidTimeZoneException("Error during SetParametersAsync");
```

Comment:
It might be good to have a comment here explaining that ErrorThrowingComponent doesn't throw if the exception is not set. It took me a while to understand the test.